packages/jest-runtime/src/index.ts

```diff
@@ -607,6 +617,9 @@ export default class Runtime {
 
   requireInternalModule<T = unknown>(from: Config.Path, to?: string): T {
     if (to) {
+      if (INTERNAL_MODULE_REQUIRE_OUTSIDE_OPTIMIZED_MODULES.has(to)) {
+        return require(to);
```

Comment:
I assume this will break for pnp - do we need to use `createRequire` on `from` or something? I'm assuming it'll resolve to the same file and still be a cache hit

Reply:
Yep that will likely happen actually. Will probably already happen with master because of the lines below this from #7792 I think?

Reply:
would be nice to verify and add a test for it

Reply:
Should be enough to enable pnp by applying https://github.com/facebook/jest/pull/9476 to this PR